tcl/filter.tcl

```diff
@@ -50,10 +50,13 @@ proc qc::filter_authenticate {event {error_handler qc::error_handler}} {
         # Check if this request is registered.
         if { [qc::registered $method $url_path] } {
             if {[qc::cookie_exists session_id] && [qc::session_valid [qc::session_id]]} {
-                qc::session_update [qc::session_id]
+                global session_id current_user_id
+                qc::session_update $session_id
+                set current_user_id [qc::session_user_id $session_id]
             } elseif {$method ni [list "GET" "HEAD"] && [qc::cookie_exists session_id] && ! [qc::session_valid [qc::session_id]]} {
```

Comment:
Calling `qc::auth` without having `current_user_id` set would run through authentication again. That's probably fine though as it likely wouldn't impact performance to any significant degree. `qc::auth` wasn't used here because we didn't want HBA for this filter although I don't recall why.

I changed it for consistency between this section and implicitly logging in the anonymous user which does set the current user ID. Now that I've looked through it setting `current_user_id` anywhere outside of `qc::auth` probably isn't necessary but the use for logging in the anonymous session had me believing that it was.